fedora_elections/elections.py

```diff
@@ -144,11 +141,19 @@ def vote_range(election):
         max_range=max_selection)
 
     if form.validate_on_submit():
-
         if form.action.data == 'submit':
             for candidate in form:
                 if candidate.short_name in ['csrf_token', 'action']:
                     continue
+                if revote:
+                    vote = update(models.Vote).\
+                    where(models.Vote.candidate_id == candidate.short_name
+                    and models.Vote.voter == flask.g.fas_user.username
+                    and models.Vote.election == election.id).\
+                    values(value = candidate.data)
+                    SESSION.execute(vote)
+                    #break out of candidate loop
+                    continue
```

Comment:
You're right the only way we can do this properly is by either:

a/ making a loop in the candidate loop to find the vote corresponding to this candidate

or 

b/ make sure the two lists (existing votes and new ones) have the same order and size before entering this loop. This way we can be sure that ``old_vote[index].candidate_id == new_vote[index].candidate_id``